detection_rules/main.py

```diff
@@ -296,3 +297,102 @@ def test_rules(ctx):
+
+    domain = original_domain.split('.')[0]
+    domain_index = f'dnstwist-{domain}'
+    # If index already exists, prompt user to confirm if they want to overwrite
+    if es_client.indices.exists(index=domain_index):
+        if click.confirm(
+            f"dnstwist index {domain_index} already exists for {original_domain}. Do you want to overwrite?", abort=True
+        ):
+            es_client.indices.delete(index=domain_index)
+
+    fields = [
+        "dns-a",
+        "dns-aaaa",
+        "dns-mx",
+        "dns-ns",
+        "banner-http",
+        "fuzzer",
+        "original-domain",
+        "dns.question.registered_domain"
+    ]
+    timestamp_field = "@timestamp"
+    mappings = {"mappings": {"properties": {f: {"type": "keyword"} for f in fields}}}
+    mappings["mappings"]["properties"][timestamp_field] = {"type": "date"}
+
+    es_client.indices.create(index=domain_index, body=mappings)
+
+    # handle dns.question.registered_domain separately
+    fields.pop()
+    es_updates = []
+    now = datetime.utcnow()
+
+    for item in dnstwist_data:
+        if item['fuzzer'] == 'original*':
+            continue
+
+        record = item.copy()
+        record.setdefault('dns', {}).setdefault('question', {}).setdefault('registered_domain', item.get('domain-name'))
+
+        for field in fields:
+            record.setdefault(field, None)
+
+        record['@timestamp'] = now
+
+        es_updates.extend([{'create': {'_index': domain_index}}, record])
+
+    click.echo(f'Indexing data for domain {original_domain}')
+
+    results = es_client.bulk(body=es_updates)
+    if results['errors']:
+        error = {r['create']['result'] for r in results['items'] if r['create']['status'] != 201}
+        client_error(f'Errors occurred during indexing:\n{error}')
+
+    # Creating threat match rule
+    root_dir = Path(__file__).parent.parent
+    rule_template_file = root_dir / 'etc' / 'rule_template_typosquatting_domain.toml'
+    custom_rule_file = (
+        root_dir / "rules" / "cross-platform" / f"initial_access_dns_request_for_typosquatting_domain_{domain}.toml"
+    )
+
+    click.echo(f'{len(results["items"])} watchlist domains identified')
+
+    click.echo('Attempting to create threat match rule')
+    template_rule = load_dump(str(rule_template_file))
+
+    # Populate template rule with user's custom info
+    today = datetime.today().strftime("%Y/%m/%d")
+    template_rule['metadata'].update(creation_date=today, updated_date=today)
+    template_rule['rule'].update(author=[author], rule_id=str(uuid4()))
+
+    # Create rule object and validate it against schema
+    rule_collection = RuleCollection()
+    rule = rule_collection.load_dict(template_rule, path=custom_rule_file)
+
+    # Save rule in toml format
+    click.echo(f'Saving rule to {custom_rule_file}')
+    rule.save_toml()
+
+    if click.confirm("Upload rule to Kibana?", abort=True):
+        ctx.invoke(upload_rule, rule_file=[custom_rule_file])
```

Comment:
@threat-punter @ajosh0504 - since there is no customization to the template rule other than a unique ID, we could just store an ndjson version of the rule and add to the docs to use the Kibana import rule functionality and override the rule ID.

This would simplify this a bit more and remove extra code

<details>
<summary>screenshot with unique ID</summary>
![image](https://user-images.githubusercontent.com/16747370/131804949-7c613ecc-8c42-48df-b6f6-a3cebce8b0ba.png)
</details>